snakemake_executor_plugin_slurm/__init__.py

```
@@ -559,10 +563,18 @@ async def job_stati(self, command):
         except subprocess.CalledProcessError as e:
-            self.logger.error(
-                f"The job status query failed with command: {command}\n"
-                f"Error message: {e.stderr.strip()}\n"
-            )
+            error_message = e.stderr.strip()
+            if "slurm_persist_conn_open_without_init" in error_message:
+                self.logger.warning(
+                    "The SLURM database might not be available yet. "
+                    "Will retry querying the job status in 5 minutes."
+                )
+                await asyncio.sleep(retry_intervals)
```

Ruff (0.8.2)

Undefined name `asyncio`

(F821)